libs/payments/events/src/lib/emitter.types.ts

```diff
@@ -13,15 +13,14 @@ export type CheckoutPaymentEvents = CommonMetrics & {
   paymentProvider?: PaymentProvidersType;
 };
 
-export type SubscriptionEnded = {
-  productId: string;
+export type SubscriptionEndedEvents = {
   priceId: string;
   priceInterval?: string;
   priceIntervalCount?: number;
   paymentProvider?: PaymentProvidersType;
   providerEventId: string;
-  subscriptionId: string;
-  voluntaryCancellation?: boolean;
+  voluntaryCancellation: boolean;
+  optedOut?: boolean;
```

Comment:
Since this is a server only event, should opt out be available for this?

Reply:
Ah, I had a look at the existing event that's currently being emitted, and yes it does look like the customers opt out preferences are being respected. So this should as well. Good catch! 👍